packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,369 @@
+}
+
+export function convertBTCToSatoshis(btc: number) {
+  return parseInt((btc * 10 ** 8).toString(), 10); // remove floating point overflow by parseInt
+}
+
+export function generateTxUniqueIdentifier(txId: string, index: number) {
+  return `${txId}:${index}`;
+}
+
+/**
+ * Converts an outpoint to id format.
+ *
+ * An outpoint refers to a specific output.
+ *
+ * Reference: https://developer.bitcoin.org/reference/transactions.html#outpoint-the-specific-part-of-a-specific-output
+ *
+ * @param outpoint Outpoint in string format `{txid}:{vout}`
+ * @returns id formatted string, `{txid}i{vout}`
+ */
+export function outpointToIdFormat(outpoint: string) {
+  if (outpoint.includes(":")) {
+    return outpoint.replace(":", "i");
+  }
+
+  return outpoint.includes("i") ? outpoint : `${outpoint}i0`;
+}
+
+export function decodePSBT({ hex, base64, buffer }: OneOfAllDataFormats): Psbt {
+  if (hex) return Psbt.fromHex(hex);
+  if (base64) return Psbt.fromBase64(base64);
+  if (buffer) return Psbt.fromBuffer(buffer);
+
+  throw new OrditSDKError("Invalid options");
+}
+
+export function decodeTx({ hex, buffer }: BufferOrHex): Transaction {
+  if (hex) return Transaction.fromHex(hex);
+  if (buffer) return Transaction.fromBuffer(buffer);
+
+  throw new OrditSDKError("Invalid options");
+}
+
+function isPaymentFactory(payment: PaymentCreator, network: Network) {
+  return (script: Buffer) => {
+    try {
+      return payment({ output: script, network: getNetwork(network) });
+    } catch (error) {
+      return false;
+    }
+  };
+}
+
+export const isP2PKH = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2pkh = isPaymentFactory(payments.p2pkh, network)(script);
+  return {
+    type: "p2pkh",
+    payload: p2pkh,
+  };
+};
+export const isP2WPKH = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2wpkh = isPaymentFactory(payments.p2wpkh, network)(script);
+  return {
+    type: "p2wpkh",
+    payload: p2wpkh,
+  };
+};
+export const isP2WSHScript = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2wsh = isPaymentFactory(payments.p2wsh, network)(script);
+  return {
+    type: "p2sh",
+    payload: p2wsh,
+  };
+};
+export const isP2SHScript = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2sh = isPaymentFactory(payments.p2sh, network)(script);
+  return {
+    type: "p2sh",
+    payload: p2sh,
+  };
+};
+export const isP2TR = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2tr = isPaymentFactory(payments.p2tr, network)(script);
+  return {
+    type: "p2tr",
+    payload: p2tr,
+  };
+};
+export function getScriptType(
+  script: Buffer,
+  network: Network,
+): GetScriptTypeResponse {
+  const p2pkh = isP2PKH(script, network);
+  if (p2pkh.payload) {
+    return {
+      format: ADDRESS_TYPE_TO_FORMAT.p2pkh,
+      ...p2pkh,
+    };
+  }
+
```

Comment:
Can we add line breaks in between functions?

Reply:
Added